### PR TITLE
WCW Automations

### PR DESCRIPTION
**Pull request by DarrylBrooks97**
[Loom Demo](https://www.loom.com/share/eff3a1f8452b4f09840463b32f6a9c19)

Includes: 
- Checking DB for WCW Topic
- Event creation
- Announcement & Thread creation

### CODE REVIEW

File: src/automations/WCW.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        const data = await prisma.wcwTopic.findFirst({
+            where: { used: false },
+        });
+
+        if (!data) {
+            const channel = client.channels.cache.get(
+                CONSTANTS.MODERATOR_CHANNEL_ID,
+            ) as TextChannel;
+            return await channel.send(
+                'All Water Cooler Wednesday topics have been used ❗️ Add a new one with /update-wcw command',
```
Comment (gt-codes) [inline src/automations/WCW.ts]: Minor formatting on the returned message 
```suggestion
                "All Water Cooler Wednesday topics have been used ❗️ Add a new one with `/update-wcw` command",
```

File: src/prisma/schema.prisma [hunk 2]
```diff
@@ -140,3 +140,9 @@ model Comment {
   @@index([authorId])
   @@index([postId])
 }
+
+model wcwTopic {
+  id    Int     @id @default(autoincrement())
+  topic String  @db.Text
+  used  Boolean
+}
```
Comment (gt-codes) [inline src/prisma/schema.prisma:150]: Add timestamp to this model. I'm just realizing that a lot of the other models don't have `createdAt` and `updatedAt` so make sure all models have these two fields.

```suggestion
  createdAt DateTime      @default(now())
  updatedAt DateTime      @updatedAt
}
```

File: src/constants.ts [hunk 3]
```diff
@@ -8,6 +8,7 @@ export const CONSTANTS = Object.freeze({
     CLIENT_ID: process.env.CLIENT_ID as string,
     GUILD_ID: process.env.GUILD_ID as string,
     GENERAL_CHANNEL_ID: process.env.GENERAL_CHANNEL_ID as string,
+    MODERATOR_CHANNEL_ID: process.env.MODERATOR_CHANNEL_ID as string,
```
Comment (gt-codes) [inline src/constants.ts:11]: Did you add this env to Railway @DarrylBrooks97 ?

**Review by gt-codes — CHANGES_REQUESTED** [review]

**Comment by DarrylBrooks97** [comment]
Closing this PR. All functionality has moved to PR #12

**Pull request closed without merging**